base/strings/string.jl

```diff
@@ -85,6 +82,36 @@ function String(v::Vector{UInt8})
     return str
 end
 
+"""
+    take_string!(v::Union{Memory{UInt8}, Vector{UInt8}}) -> String
+
+Create a string, truncating `v` to zero length. If `v` is a `Vector`, further
+modification of `v` will not modify the string.
+When possible, the returned string will reuse the memory of `v`.
```

Comment:
Normally `Memory` length can't be mutated, so it might not get truncated. This should also 
 say mutating the `Memory` after taking a string might be unsafe.

Reply:
This implementation will truncate the memory to size zero, which means it's impossible to mutate the memory.

Reply:
`jl_genericmemory_to_string` is going to change to not truncate the memory:
https://github.com/JuliaLang/julia/pull/54372#issuecomment-2102885169